catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/CswEndpoint.java

```diff
@@ -1323,6 +1342,24 @@ private CswRecordCollection queryProductById(String id) throws CswException {
         return cswRecordCollection;
     }
 
+    protected long getRange(String rangeValue) throws UnsupportedQueryException {
+        long response = -1;
+
+        if (StringUtils.isNotBlank(rangeValue)) {
+            if (rangeValue.startsWith(CswConstants.BYTES_EQUAL)) {
+                String tempString = rangeValue.substring(CswConstants.BYTES_EQUAL.length());
+                List<String> range = Splitter.on('-')
+                        .splitToList(tempString);
+                if (!range.isEmpty() && range.size() <= 2) {
```

Comment:
So an invalid range like 123-345-888 will result in a return of -1? I like that.